beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go

```diff
@@ -48,19 +48,41 @@ func (vs *Server) setExecutionData(ctx context.Context, blk interfaces.SignedBea
 	if err != nil {
 		log.WithError(err).Warn("Proposer: failed to check if builder can be used")
 	} else if canUseBuilder {
-		h, err := vs.getPayloadHeaderFromBuilder(ctx, slot, idx)
+		builderPayload, err := vs.getPayloadHeaderFromBuilder(ctx, slot, idx)
 		if err != nil {
 			builderGetPayloadMissCount.Inc()
 			log.WithError(err).Warn("Proposer: failed to get payload header from builder")
 		} else {
-			blk.SetBlinded(true)
-			if err := blk.SetExecution(h); err != nil {
-				log.WithError(err).Warn("Proposer: failed to set execution payload")
+			// Can only check values if block is Capella or later.
+			if blk.Version() >= version.Capella {
+				localPayload, err := vs.getExecutionPayload(ctx, slot, idx, blk.Block().ParentRoot(), headState)
+				if err != nil {
+					return errors.Wrap(err, "failed to get execution payload")
+				}
+				localValue, err := localPayload.Value()
+				if err != nil {
+					return errors.Wrap(err, "failed to get local payload value")
+				}
+				// Compare payload values between local and builder. Default to the local value if it is higher.
+				builderValue, err := builderPayload.Value()
+				if err != nil {
+					log.WithError(err).Warn("Proposer: failed to get builder payload value") // Default to local if can't get builder value.
+				} else if builderValue.Cmp(localValue) > 0 {
```

Comment:
To me, big int comparison methods are always unintuitive and I wish I could remember the < and > conditions. Maybe a comment ? Perhaps we could benefit instead from a `Max` method that acts on big ints because thats what we want

Reply:
I'll add a comment